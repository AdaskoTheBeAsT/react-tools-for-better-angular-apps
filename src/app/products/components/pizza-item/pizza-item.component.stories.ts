import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { storiesOf, moduleMetadata } from '@storybook/angular';
import {
  withKnobs,
  text,
  number,
  boolean,
  array,
  select,
  color,
  date,
  button,
} from '@storybook/addon-knobs/angular';

import { StylesComponent } from '../../../../stories/module-metadata';
import { PizzaItemComponent } from './pizza-item.component';
import { PizzaDisplayComponent } from '../pizza-display/pizza-display.component';
import { pizzaMock, toppingsMock } from './pizza-item.mocks';

storiesOf('Pizza Item', module)
  .addDecorator(withKnobs)
  .addDecorator(
    moduleMetadata({
      imports: [BrowserAnimationsModule, RouterTestingModule],
      declarations: [PizzaItemComponent, PizzaDisplayComponent, StylesComponent],
      providers: [],
    })
  )
  .add('render empty pizza', () => {
    return {
      template: `
        <storybook-styles>
          <pizza-item [pizza]="pizza"></pizza-item>
        </storybook-styles>
      `,
      props: {
        pizza: {},
      },
    };
  })
  .add('render pizza without toppings!', () => {
    return {
      template: `
        <storybook-styles>
          <pizza-item [pizza]="pizza"></pizza-item>
        </storybook-styles>
      `,
      props: {
        pizza: pizzaMock,
      },
    };
  })
  .add('render pizza with toppings!', () => {
    return {
      template: `
        <storybook-styles>
          <pizza-item [pizza]="pizza"></pizza-item>
        </storybook-styles>
      `,
      props: {
        pizza: { ...pizzaMock, toppings: toppingsMock },
      },
    };
  });
