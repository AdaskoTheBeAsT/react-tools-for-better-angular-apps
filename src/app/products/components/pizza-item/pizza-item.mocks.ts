import { Pizza, Topping } from '../../models';

export const toppingsMock: Topping[] = [
  {
    id: 1,
    name: 'anchovy',
  },
  {
    id: 2,
    name: 'bacon',
  },
];
export const pizzaMock: Pizza = {
  id: 1,
  name: 'Pepperoni pizza',
  toppings: [],
};
