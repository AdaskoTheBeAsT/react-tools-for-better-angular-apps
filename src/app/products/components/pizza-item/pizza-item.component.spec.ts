import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { Pizza, Topping } from '../../models';
import { PizzaItemComponent } from './pizza-item.component';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe(`${PizzaItemComponent.name}`, () => {
  let component: PizzaItemComponent;
  let fixture: ComponentFixture<PizzaItemComponent>;
  const toppingsMock: Topping[] = [
    {
      id: 1,
      name: 'anchovy',
    },
    {
      id: 2,
      name: 'bacon',
    },
  ];
  const pizzaMock: Pizza = {
    id: 1,
    name: 'Pepperoni pizza',
    toppings: [],
  };

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        imports: [RouterTestingModule],
        declarations: [PizzaItemComponent],
        schemas: [NO_ERRORS_SCHEMA],
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(PizzaItemComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it(`should display pizza toppings`, () => {
    component.pizza = pizzaMock;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
