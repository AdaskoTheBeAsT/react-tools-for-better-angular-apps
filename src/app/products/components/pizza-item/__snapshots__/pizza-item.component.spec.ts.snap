// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaItemComponent should display pizza toppings 1`] = `
<pizza-item
  pizza={[Function Object]}
>
  
      <div
    class="pizza-item"
  >
    
        
    <a
      href="/products/undefined"
      ng-reflect-router-link="/products,"
    >
      
          
      <pizza-display />
      
          
      <h4>
        
      </h4>
      
          
      <button
        class="btn btn__ok"
        type="button"
      >
        
            View Pizza
          
      </button>
      
        
    </a>
    
      
  </div>
    
</pizza-item>
`;
